core/src/main/scala/dagr/core/execsystem/TaskExecutionInfo.scala

```diff
@@ -42,7 +42,6 @@ import dagr.core.tasksystem.Task
  * @param attemptIndex the one-based count of attempts to run this task.
  */
 class TaskExecutionInfo(var task: Task,
-                        val id: BigInt,
                         var status: TaskStatus.Value,
```

Comment:
I wanted task at being good at the execution part, not the tracking part, hence another class to store all of this info.  You could ask the same question about the other info too.  If we go down the route of putting more of this info into task, I would suggest having task store a reference to a `TaskExecutionInfo`.